src/utils.mli

```diff
@@ -53,6 +53,9 @@ module Ast_io : sig
       -> (t, read_error) result
 
     val write : out_channel -> t -> add_ppx_context:bool -> unit
+    module Read_bin : sig 
+      val read : string ->(t, read_error) result
+    end 
```

Comment:
Since you only expose `Read_bin` in ppxlib's public API and not its ambient module `Ast_io` (which I think is the right thing to do), the return type of `read` won't form part of the public API. So it won't be possible to handle the return value of `read` (unless folks make use of our private API, which would be good to avoid). So we have to think about what to do with the two types `Ast_io.t` and `Ast_io.read_error` that form part of the return type.

About `Ast_io.t`: What do you think about re-exposing `Ast_io.t` inside `Read_bin` as an abstract type and adding getter functions for `input_name` and for `ast`? For the latter you could choose a good way to express if a signature or a structure is returned, maybe there could be two functions `read_sig` and `read_str` or similar?

About `Ast_io.read_error`: that type is a bit ppxlib internal specific. What do you think about returning a string for an error instead of a `read_error`, in which you describe in words what's gone wrong?